compiler/rustc_mir/src/util/pretty.rs

```diff
@@ -289,19 +289,29 @@ pub fn write_mir_pretty<'tcx>(
             }
             Ok(())
         };
-        match tcx.hir().body_const_context(def_id.expect_local()) {
-            None => render_body(w, tcx.optimized_mir(def_id))?,
-            // For `const fn` we want to render the optimized MIR. If you want the mir used in
-            // ctfe, you can dump the MIR after the `Deaggregator` optimization pass.
-            Some(rustc_hir::ConstContext::ConstFn) => {
-                render_body(w, tcx.optimized_mir(def_id))?;
-                writeln!(w)?;
-                writeln!(w, "// MIR FOR CTFE")?;
-                // Do not use `render_body`, as that would render the promoteds again, but these
-                // are shared between mir_for_ctfe and optimized_mir
-                write_mir_fn(tcx, tcx.mir_for_ctfe(def_id), &mut |_, _| Ok(()), w)?;
+
+        // For `const fn` we want to render the optimized MIR. If you want the mir used in
+        // ctfe, you can dump the MIR after the `Deaggregator` optimization pass.
```

Comment:
THis comment is outdated (it was before this PR, I forgot to update it). It should say that for `const fn` we want to render both the `optimized_mir` and the `mir_for_ctfe`.

Reply:
Updated the comment.